prefab/README.md

```diff
@@ -0,0 +1,90 @@
+Prefab Samples
+==============
+
+The samples here are to demontrate how to use [the Prefab feature] in Android Gradle Plugin.
+Generally speaking, Prefab is a mechanism to make C/C++ libraries available to the gradle's
+C/C++ build system, with the requirement that the C/C++ libraries are organized in the
+[Prefab format] and packed inside AARs. In order to use the C/C++ dependency AAR, as AAR authors,
+you need to do the following:
+
+1. Use AGP to pack C/C++ libraries (including the header files), called modules, into AAR
+1. Distribute AARs with [Google Maven] or other mechanisms, but this is not Prefab specific
+1. Add AAR dependency to your application, now you have access to the AAR (but not prefab modules yet);
+   again this is not Prefab specific either
+1. Enable Prefab in the gradle scripts to access prefab module(s) inside AAR
+1. Use Prefab modules in your native build system such as CMake or ndk-build
+
+However as AAR users, you just need to do step 3 through step 5 above, i.e:
+1. Add AAR dependency to your application, now you have access to the AAR (but not prefab modules yet);
+   again this is not Prefab specific either
+1. Enable Prefab in the gradle scripts to access prefab module(s) inside AAR
+1. Use Prefab modules in your native build system such as CMake or ndk-build
+
+Note that the step 2 and 3 above are not C/C++ or prefab specific, please refer to
+[the AAR documentation] for details.
```

Comment:
The samples in this directory demonstrate how to create and consume C/C++
dependencies with the Android Gradle Plugin. Dependencies are packaged within
AARs using the [Prefab] package format. The Android Gradle Plugin natively
supports producing and consuming these, so you do not need to worry about the
details of the packaging format.

The individual samples will explain in more detail, but the typical workflow for
a consumer of a native dependency is:

1. Enable the `prefab` build feature in your `build.gradle` file.
2. Add the dependency to the `dependencies` block of your build.gradle file.
3. Import the package in your `CMakeLists.txt` or `Android.mk` file.
4. Link the dependencies to your libraries.

Linking the dependency to your library will automatically make the headers
available, link the required libraries, and include the dependencies you use in
your APK or AAR.

To produce an AAR that exposes C/C++ APIs:

1. Enable the `prefabPublishing` build feature in your `build.gradle` file.
2. Use `android.prefab` in your `build.gradle` file to declare the names of the
   libraries you wish to export and the headers that define their interface.

To test your packages, follow the steps for package consumers. Note that until
https://issuetracker.google.com/120166563 is fixed it is not possible to depend
on the native components of an AAR generated by another module of the same
project when using Android Studio. Instead, test in a separate project. You can
import the AAR into your test project either [directly] or by publishing it to a
Maven repository.

[directly]: https://developer.android.com/studio/projects/android-library#AddDependency
[Prefab]: https://google.github.io/prefab

Reply:
done